testing/tools/integration/integration.py

```diff
@@ -1333,4 +1357,6 @@ def pipeline_test(generator, expected, command, workers=1, sources=1,
                                             outputs))
 
     # Return runner names and outputs if try block didn't have a return
-    return [(r.name, r.get_output()) for r in runners]
+    return_value = [(r.name, r.get_output()) for r in runners]
+    clean_resilience_path(resilience_dir)
```

Comment:
This should be in the `finally` block if it needs to be performed regardless of exceptions that might be raised in the test's main `try.. .except` scope. In that case I would suggest doing it before `if alive:`  test.

Reply:
The Python docs say that exceptions will be discarded if a `finally` block executes a `return`, so I went a conservative route and used commit 4a3c677 to alter execution flow.